scripts/translation/xliffmerger.py

```diff
@@ -0,0 +1,95 @@
+    xml_tree_translation = ET.parse(translation_file)
+    original_body_element, original_trans_units = get_trans_units(xml_tree_original)
+    translation_body_element, translation_trans_units = get_trans_units(xml_tree_translation)
+    logging.debug(f'{len(original_trans_units)} trans_units in original file {original_file}')
+    logging.debug(f'{len(translation_trans_units)} trans_units in translation file {translation_file}')
+
+    org_trans_units_ids = [tr_unit.attrib['id'] for tr_unit in original_trans_units]
+    trans_units_ids = [tr_unit.attrib['id'] for tr_unit in translation_trans_units]
+    logging.debug(f'{org_trans_units_ids} trans_units IDs in original file {original_file}')
+    logging.debug(f'{trans_units_ids} trans_units in Ids translation file {translation_file}')
+
+    # Delete the ones which are in translation file but not in original
+    logging.info("########## DELETING ORPHAN TRANS_UNITS ##########")
+    to_remove = set(trans_units_ids).difference(set(org_trans_units_ids))
+    logging.info(f'Trans-units with these Ids : {to_remove}, will be deleted from the {translation_file}')
+    delete_trans_units(translation_body_element, translation_trans_units, to_remove)
+    logging.info("-------------------------------------------------\n")
+
+    # Add the ones which are not in translation file but exist in original
+    logging.info("########## ADDING MISSING TRANS_UNITS ##########")
+    to_add = set(org_trans_units_ids).difference(set(trans_units_ids))
+    logging.info(f'Trans-units with these Ids : {to_add}, will be added to the {translation_file}')
+    add_trans_units(translation_body_element, original_trans_units, to_add)
+    logging.info("-------------------------------------------------\n")
+
+    # Update() those trans_units where source has been changed but id remained same. We will be updating source text and
+    # add the 'needs-adaptation' state attribute
+    logging.info("########## UPDATE THE CHANGED TRANS_UNITS ##########")
+    # Get again so we get the updated list after deletion/addition
+    translation_trans_units = list(translation_body_element)
+
+    logging.debug(f'{len(original_trans_units)} trans_units in original file {original_file}')
+    logging.debug(f'{len(translation_trans_units)} trans_units in original file {original_file}')
+    if len(translation_trans_units) == len(original_trans_units):
+        trans_units_sources = {tr_unit.attrib['id']: tr_unit[0].text for tr_unit in translation_trans_units}
+        org_trans_units_sources = {tr_unit.attrib['id']: tr_unit[0].text for tr_unit in original_trans_units}
+        to_update = {k: org_trans_units_sources[k] for k,_ in set(org_trans_units_sources.items()) - set(trans_units_sources.items())}
+        logging.info(f' These (id,new_source_value) -> {to_update}, will be updated in the {translation_file} ')
+        update_trans_units(translation_body_element,translation_trans_units,to_update)
+
```

Comment:
What about `else:` (if len(translation_trans_units) == len(original_trans_units):)

I think we should at least write a message